specification/sbi_atee.adoc

```diff
@@ -93,67 +93,151 @@ The host must ensure that the TSM has sufficient confidential memory for mapping
-`sbi_tee_host_convert_pages()`, or by using converted page that aren't currently assigned to a TVM. The TSM copies the 
+`sbi_tee_host_convert_pages()`, or by using converted pages that aren't currently assigned to a TVM. The TSM copies the 
 payload for the TVM from non-confidential pages to confidential pages, and extends the corresponding measurements for the
 TVM.
 
-===== VCPU shared state enumeration
-The TSM communicates additional information about TVM exits from `sbi_tee_host_run_tvm_vcpu()` using a non-confidential
-shared memory region that's configured on a per-VCPU basis by the host. The host can also use this shared memory
-region to control and configure TVM parameters like the initial-entry point (SEPC), initial parameter, etc., and
-to respond to TVM exits.
+===== VCPU shared state
+Host needs access to some of the TVM CSRS and GPRs to handle TVM exits. For example, the host needs `htval` to determine
+the fault address, `a0`-`a7` GPRs are needed to handle forwarded ECALLs and so on. For this purpose, the host and TSM use NACL
+Extension based shared memory interface <<R10>>, from now on called NACL shared memory to avoid confusion with shared memory pages
+between TVM and the host.
 
-The layout of the shared-memory region can vary by TSM version. The host can determine the size and offset of the
-regions enumerated in `vcpu_register_set_id` by calling `sbi_tee_host_get_tvm_vcpu_num_register_sets()` to get the
-number of enumerated sets, and then `sbi_tee_host_get_tvm_vcpu_num_register_sets()` to determine the offset.
+The NACL shared memory interface is between TSM and the host and TSM is responsible for writing any trap-related CSRs and GPRs
+needed by the host to handle the exception. TSM is also responsible for reading the returned result and forwarding it to the TVM.
+Further details about which CSRs and GPRS are used by the TSM and the host can be found in <<table_tsm_csr_updates_in_nacl>>.
+The layout of NACL shared memory is shown below as `struct nacl_shmem` and `scratch` space layout for TSM is shown as
+`struct tsm_shmem_scratch`.
+
+[source, C]
+-------------------
+struct nacl_shmem {
+    /* Scratch space. The layout of this scratch space is defined by the particular function being
+     * invoked.
+     *
+     * For the `sbi_tee_host_run_tvm_vcpu()` function in the TEE-H extension, the layout of this 
+     * scratch space matches the `tsm_shmem_scratch` struct given below.
+     */
+    uint64_t scratch[256];
+    uint64_t _reserved[240];
+    /* Bitmap indicating which CSRs in `csrs` the host wishes to sync.
+     *
+     * Currently unused in the TEE-related extensions and will not be read or written by the TSM.
+     */
+    uint64_t dirty_bitmap[16];
+    /* Hypervisor and virtual-supervisor CSRs. The 12-bit CSR number is transformed into a 10-bit
+     * index by extracting bits `{csr[11:10], csr[8:0]}` since `csr[9:8]` is always 2'b10 for HS
```

Comment:
should this be `csr[11:10], csr[7:0]`

Reply:
Nice catch. Thanks. Fixed.
Also rebased the branch on main and updated the pdf snapshot.